components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Connection.kt

```diff
@@ -87,6 +106,12 @@ internal object RustPlacesConnection : Connection {
         return cachedReader!!
     }
 
+    override fun newReader(): PlacesReaderConnection = synchronized(this) {
```

Comment:
Went with adding support for new readers since I saw them being used for multiple scenarios.
One separate reader used only for the history data, that we have full control over seemed safer.